Crash when navigating back after opening any link on login flow

**Describe the bug**
Crash when navigating back after opening any link on login flow. 

**IMPORTANT NOTE** Only happens when the developer setting "Don't keep activities" is enabled

**To Reproduce**
Issue related with Android navigation changes added here #7534

**Screenshots**

https://user-images.githubusercontent.com/2663464/196935409-928035ad-a02c-47e7-9a69-d4249d16d5b1.mp4

**Expected behaviour**
Don't crash when navigating back from any of the opened links

**Stacktrace of crash**

```
FATAL EXCEPTION: main
                                                                                                    Process: com.woocommerce.android.dev, PID: 25206
                                                                                                    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.woocommerce.android.dev/com.woocommerce.android.ui.login.LoginActivity}: java.lang.ClassCastException: com.woocommerce.android.ui.login.overrides.WooLoginEmailFragment cannot be cast to androidx.navigation.fragment.NavHostFragment
                                                                                                    	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3449)
                                                                                                    	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3601)
                                                                                                    	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                                                    	at android.os.Looper.loop(Looper.java:223)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7656)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
                                                                                                    Caused by: java.lang.ClassCastException: com.woocommerce.android.ui.login.overrides.WooLoginEmailFragment cannot be cast to androidx.navigation.fragment.NavHostFragment
                                                                                                    	at com.woocommerce.android.ui.login.LoginActivity.onCreate(LoginActivity.kt:180)
                                                                                                    	at android.app.Activity.performCreate(Activity.java:7994)
                                                                                                    	at android.app.Activity.performCreate(Activity.java:7978)
                                                                                                    	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
                                                                                                    	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3422)
                                                                                                    	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3601) 
                                                                                                    	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85) 
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
                                                                                                    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066) 
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106) 
                                                                                                    	at android.os.Looper.loop(Looper.java:223) 
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7656) 
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592) 
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947) 
```

--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginActivity.kt
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginActivity.kt
@@ -155,9 +155,9 @@ class LoginActivity :
 
     private var loginMode: LoginMode? = null
     private lateinit var binding: ActivityLoginBinding
-    private val navController: NavController by lazy {
-        navHostFragment.navController
-    }
+
+    // Lazy init required to avoid crash. Full context: github.com/woocommerce/woocommerce-android/pull/7581
+    private val navController: NavController by lazy { navHostFragment.navController }
     private val navHostFragment: NavHostFragment by lazy {
         supportFragmentManager.findFragmentById(R.id.nav_host_fragment_login) as NavHostFragment
     }

--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginActivity.kt
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginActivity.kt
@@ -155,8 +155,12 @@ class LoginActivity :
 
     private var loginMode: LoginMode? = null
     private lateinit var binding: ActivityLoginBinding
-    private lateinit var navController: NavController
-    private lateinit var navHostFragment: NavHostFragment
+    private val navController: NavController by lazy {
+        navHostFragment.navController
+    }
+    private val navHostFragment: NavHostFragment by lazy {
+        supportFragmentManager.findFragmentById(R.id.nav_host_fragment_login) as NavHostFragment
+    }
 
     private var connectSiteInfo: ConnectSiteInfo? = null
 
@@ -177,8 +181,6 @@ class LoginActivity :
 
         binding = ActivityLoginBinding.inflate(layoutInflater)
         setContentView(binding.root)
-        navHostFragment = supportFragmentManager.findFragmentById(R.id.nav_host_fragment_login) as NavHostFragment
-        navController = navHostFragment.navController
 
         val loginHelpNotification = getLoginHelpNotification()
 

